tez-mapreduce/src/main/java/org/apache/tez/mapreduce/committer/MROutputCommitter.java

```diff
@@ -119,6 +120,7 @@ public void abortOutput(VertexStatus.State finalState) throws IOException {
         || jobConf.getBoolean("mapred.mapper.new-api", false))  {
       newApiCommitter = true;
     }
+    jobConf.set(MRJobConfig.MR_JOB_UUID, Utils.createJobUUID(getContext().getApplicationId().getClusterTimestamp(), getContext().getApplicationId().getId(), getContext().getDagIdentifier()));
```

Comment:
[Question] Shouldn't we be doing this in the `initialize` method ?

Reply:
Moved code to initialize.